gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaSource.java

```diff
@@ -451,7 +459,43 @@ private WorkUnit getWorkUnitForTopicPartition(KafkaPartition partition, SourceSt
         LOG.warn(
             offsetNotFoundMsg + "This partition will start from the earliest offset: " + offsets.getEarliestOffset());
         offsets.startAtEarliestOffset();
-      } else {
+      } else if (offsetOption.equals(OFFSET_LOOKBACK)) {
+        long lookbackOffsetRange = state.getPropAsLong(KAFKA_OFFSET_LOOKBACK , 0L);
+        long offset = offsets.getLatestOffset() - lookbackOffsetRange;
+        LOG.warn(offsetNotFoundMsg + "This partition will start from latest-lookback [ " + offsets.getLatestOffset() + " - " + lookbackOffsetRange + " ]  start offset: " + offset);
```

Comment:
You could have an incorrect log line due to a race condition: offsets.getLatestOffset() here could be different from the result on line 464. I'd suggest caching the value in a variable (on line 464) and using it in subsequent code.

Reply:
Sure Thanks .. I have updated this line to store offsets.getLatestOffset() in a variable  and use that in subsequent code. But I am finding same issue in multiple places like here https://github.com/apache/incubator-gobblin/pull/2721/files#diff-1f786615c11702cd74b076ab157da963R479 ... Which might lead to incorrect logging and using different offset in "offsets.startAtLatestOffset();". Please suggest something here.